### PR TITLE
2436 add 3d visualization settings

### PR DESCRIPTION
**Pull request by Corwin-Kh**

### CODE REVIEW

File: OsmAnd/res/values/strings.xml [hunk 1]
```diff
@@ -13,6 +13,8 @@
     <string name="map_widget_map_scale">Map scale</string>
     <string name="upload_local_versions">Upload local versions</string>
     <string name="download_cloud_versions">Download cloud versions</string>
+    <string name="get">Get</string>
```
Comment (Chumva) [inline OsmAnd/res/values/strings.xml:16]: shared_string_get ?

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 2]
```diff
@@ -1195,11 +1195,11 @@ public List<Amenity> searchAmenities(SearchPoiTypeFilter filter, double topLatit
 						break;
 					}
 					if (index != null && index.checkContainsInt(top31, left31, bottom31, right31)) {
-						List<Amenity> r = index.searchAmenities(top31,
-								left31, bottom31, right31, zoom, filter, matcher);
-						if (r != null) {
-							amenities.addAll(r);
-						}
+//						List<Amenity> r = index.searchAmenities(top31,
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1198]: Is it ok to disable this?

File: OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java [hunk 3]
```diff
@@ -24,7 +27,7 @@
 import java.util.concurrent.ConcurrentLinkedQueue;
 
 public class GpxDbHelper implements GpxDbReaderCallback {
-
+	public static final Log log = PlatformUtil.getLog(GpxDbHelper.class);
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java:30]: If we use this log only in this class => private?

File: OsmAnd/src/net/osmand/plus/routepreparationmenu/cards/ReverseTrackCard.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 import net.osmand.plus.routepreparationmenu.RoutingOptionsHelper.LocalRoutingParameter;
 import net.osmand.plus.routepreparationmenu.RoutingOptionsHelper.OtherLocalRoutingParameter;
 
-public class ReverseTrackCard extends MapBaseCard {
+public class ReverseTrackCard extends BaseCard {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/routepreparationmenu/cards/ReverseTrackCard.java:15]: Why we changed parent class?

File: OsmAnd/src/net/osmand/plus/views/layers/geometry/RouteGeometryWay.java [hunk 5]
```diff
@@ -159,8 +161,10 @@ protected boolean addInitialPoint(RotatedTileBox tb,
 	protected List<List<DrawPathData31>> cutStartOfCachedPath(@NonNull MapRendererView mapRenderer,
 	                                                          @NonNull RotatedTileBox tb,
 	                                                          int startLocationIndex,
-	                                                          boolean previousVisible) {
-		List<List<DrawPathData31>> croppedPathData31 = super.cutStartOfCachedPath(mapRenderer, tb, startLocationIndex, previousVisible);
+	                                                          boolean previousVisible,
+	                                                          boolean use3DVisualization
+	                                                          ) {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/geometry/RouteGeometryWay.java:166]: Formatting?

File: OsmAnd/res/layout/bottom_sheet_item_with_switch_blocked_for_free.xml [hunk 6]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Chumva) [inline OsmAnd/res/layout/bottom_sheet_item_with_switch_blocked_for_free.xml:2]: bottom_sheet_item_with_switch_and_button?

File: OsmAnd/src/net/osmand/plus/track/cards/BaseSwitchCard.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.fragment.app.FragmentActivity;
+
+import net.osmand.plus.R;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.routepreparationmenu.cards.BaseCard;
+
+public abstract class BaseSwitchCard extends BaseCard {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/track/cards/BaseSwitchCard.java:16]: DirectionArrowsCard & ShowStartFinishCard can also use this as their parent

File: OsmAnd/src/net/osmand/plus/track/cards/Track3DCard.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+		super.updateContent();
+
+		View getBtn = view.findViewById(R.id.get_btn);
+		AndroidUiHelper.updateVisibility(getBtn, isGetBtnVisible());
+		getBtn.setOnClickListener((v) -> openChoosePlan());
+	}
+
+	private boolean isGetBtnVisible() {
+		boolean isFullVersion = !Version.isFreeVersion(app) || InAppPurchaseUtils.isFullVersionAvailable(app, false);
+		return !isFullVersion && !InAppPurchaseUtils.isSubscribedToAny(app, false);
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/track/cards/Track3DCard.java:56]: Is this correct or should we check InAppPurchaseUtils.is3dMapsAvailable(app)?

File: OsmAnd/src/net/osmand/plus/views/layers/geometry/MultiProfileGeometryWay.java [hunk 9]
```diff
@@ -45,9 +45,9 @@ public MultiProfileGeometryWay(MultiProfileGeometryWayContext context) {
 		super(context, new MultiProfileGeometryWayDrawer(context));
 	}
 
-	public void drawSegments(Canvas canvas, RotatedTileBox tileBox) {
+	public void drawSegments(Canvas canvas, RotatedTileBox tileBox, boolean use3DVisualization) {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/geometry/MultiProfileGeometryWay.java:48]: Do we need to pass the use3DVisualization flag in all these methods? We can probably try to do similar to GpxGeometryWay.drawDirectionArrows

File: OsmAnd/src/net/osmand/plus/views/layers/geometry/GpxGeometryWay.java [hunk 10]
```diff
@@ -192,4 +208,25 @@ public void clearWay() {
 			super.clearWay();
 		}
 	}
+
+	@Override
+	public void calculatePath(@NonNull List<Integer> indexes, @NonNull List<Integer> xs, @NonNull List<Integer> ys, @Nullable List<GeometryWayStyle<?>> styles, @NonNull List<GeometryWayDrawer.DrawPathData31> pathsData) {
+		super.calculatePath(indexes, xs, ys, styles, pathsData);
+		for (int i = 0; i < pathsData.size(); i++) {
+			GeometryWayDrawer.DrawPathData31 drawPathData = pathsData.get(i);
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/geometry/GpxGeometryWay.java:216]: I am not sure if it is ok to calculate points' height here, or if we should calculate them similarly to cache arrays in GeometryWay

File: OsmAnd/res/layout/bottom_sheet_item_with_switch_blocked_for_free.xml (hunk 6 above)
Comment (Corwin-Kh) [reply OsmAnd/res/layout/bottom_sheet_item_with_switch_blocked_for_free.xml:2]: they are not the same. we have compound button with icon

File: OsmAnd/src/net/osmand/plus/track/cards/Track3DCard.java (hunk 8 above)
Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/track/cards/Track3DCard.java:56]: feature available for any purchase ("Available only for paid customers")
in plans it should not be connected with 3DRelief feature

File: OsmAnd/src/net/osmand/plus/views/layers/geometry/RouteGeometryWay.java (hunk 5 above)
Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/views/layers/geometry/RouteGeometryWay.java]: it's the same as the other params

**Pull request merged**